backend/inspirehep/records/api/base.py

```diff
@@ -176,6 +178,18 @@ def get_record(cls, record_uuid, with_deleted=False, record_version=None):
             raise NoResultFound
         return record
 
+    @classmethod
+    def get_records_batched(cls, ids, with_deleted=False, max_batch=100):
+        for batch in chunker(ids, max_chunk_size=max_batch):
+            query = cls.model_cls.query.filter(cls.model_cls.id.in_(batch))
+            if not with_deleted:
+                query = query.filter(
+                    (cls.model_cls.json.op("->")("deleted") == None)  # noqa
+                    | (cls.model_cls.json["deleted"] != cast("True", JSONB))
```

Comment:
I believe the first line is not needed here, as Postgres puts `NULL` when the key is absent.

Reply:
It is needed, tests don't pass when it's not there (the only result is the record with `deleted:False`)